### PR TITLE
feat(textfield): add read-only state

### PR DESCRIPTION
**Pull request by pfulton**
In order to support future use cases, this adds and styles a `readonly` state to the textfield component.

It ultimately addresses the request in [CSS-30 (add readonly state to Datepicker)](https://jira.corp.adobe.com/secure/RapidBoard.jspa?rapidView=33123&projectKey=CSS&view=detail&selectedIssue=CSS-30) and adds an example to the Datepicker's documentation to show the proper markup to achieve a `readonly` Datepicker.

Additionally, this fixes two browser console errors where we were attempting to manipulate classes for elements that did not always exist.

URL for validation:
https://git.corp.adobe.com/pages/pfulton/spectrum-css-ccx-verify/docs/textfield.html

## How and where has this been tested?
 - **How this was tested:** 
 - **Browser(s) and OS(s) this was tested with:** 

## To-do list

- [ ] If my change impacts other components, I have tested to make sure they don't break.
- [ ] If my change impacts documentation, I have updated the documentation accordingly.
- [ ] I have read the [CONTRIBUTING document](/.github/CONTRIBUTING.md).

- [ ] This pull request is ready to merge.

### CODE REVIEW

**Comment by pfulton** [comment]
This has been verified/approved by design (Jacob)

File: site/resources/js/enhancement.js [hunk 1]
```diff
@@ -469,6 +469,7 @@ function toggleInputGroupFocus(event) {
   var classList = event.target.classList;
   var closestSelector;
   // target within InputGroup
+  if (!classList) return;
```
Comment (lazd) [inline site/resources/js/enhancement.js:472]: Strange, why would this ever happen?

**Comment by lazd** [comment]
LGTM, note that this has left padding when read-only, which I think is correct, but does not match the designs we looked at the other day in Engineering Workshop (there was no left padding, I remember complaining about it).

File: site/resources/js/enhancement.js (hunk 1 above)
Comment (pfulton) [reply site/resources/js/enhancement.js:472]: FWIW, it's happening in production on any page.

**Comment by pfulton** [comment]
> LGTM, note that this has left padding when read-only, which I think is correct, but does not match the designs we looked at the other day in Engineering Workshop (there was no left padding, I remember complaining about it).

Great call-out. I'll double-check with Jacob once more to be sure.

**Pull request merged**